browser/resources/settings/brave_settings_overrides.js

```diff
@@ -1,9 +1,317 @@
+// Copyright (c) 2019 The Brave Authors. All rights reserved.
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at http://mozilla.org/MPL/2.0/.
 
-window.BravePatching.registerPolymerBehaviors({
+// Local Scope
+(function() {
+// TODO: move to a module
+function throttle (callback, maxWaitTime = 30) {
+  // Call on first invocation
+  let shouldWait = false;
+  return function (...args) {
+    if (!shouldWait) {
+      callback.apply(this, args);
+      shouldWait = true;
+      setTimeout(function () {
+        shouldWait = false;
+      }, maxWaitTime);
+    }
+  }
+}
+
+// Utils
+function createMenuElement (title, href, iconName) {
+  const menuEl = document.createElement('a')
+  menuEl.href = href
+  menuEl.innerHTML = `
+    <iron-icon icon="${iconName}"></iron-icon>
+    ${title}
+  `
+  return menuEl
+}
+
+function getMenuElement (templateContent, href) {
+  const menuEl = templateContent.querySelector(`a[href="${href}"]`)
+  if (!menuEl) {
+    console.error(`[Brave Settings Overrides] Could not find menu item '${href}'`)
+  }
+  return menuEl
+}
+
+function getSectionElement (templateContent, sectionName) {
+  const sectionEl = templateContent.querySelector(`template[if="[[showPage_(pageVisibility.${sectionName})]]"]`)
+  if (!sectionEl) {
+    console.error(`[Brave Settings Overrides] Could not find section '${sectionName}'`)
+  }
+  return sectionEl
+}
+
+if (!BravePatching) {
+  console.error('BravePatching was not available to brave_settings_overrides.js')
+}
+
+//
+// Override, extend or modify existing modules
+//
+
+const BraveClearSettingsMenuHighlightBehavior = {
+  ready: function() {
+    // Clear menu selection after scrolling away.
+    // Chromium's menu is not persistant, so does not have
+    // this issue.
+    const container = this.$.container
+    if (!container) {
+      console.error('Could not find #container in settings-ui module')
+    }
+    const menu = this.$$('settings-menu')
+    if (!menu) {
+      console.error('Could not find settings-menu in settings-ui module')
+    }
+    let onScroll
+    function stopObservingScroll() {
+      if (onScroll) {
+        container.removeEventListener('scroll', onScroll)
+        onScroll = null
+      }
+    }
+    window.addEventListener('showing-section', ({ detail: section }) => {
+      // Currently showing or about to scroll to `section`.
+      // If we're getting further away from section top
+      // then section is no longer 'selected'.
+      // TODO(petemill): If this wasn't a chromium module, we'd simply add a handler
+      // for scrolling away, or have the menu change selection as we scroll.
+      stopObservingScroll()
```

Comment:
curious about this code... let's say I click `Get Started`. Sidebar properly shows it selected (orange). As I scroll down, it does "go out of scope" and un-select itself. But I'm not seeing `Extensions`, `Sync`, `Appearance`, etc select themselves. Is the highlighted section supposed to be shown? (also, if so- maybe we can call out in test plan)

Reply:
@bsclifton no support for scrolling as you move. Just kept the current chromium highlighting behavior and removed it when scrolling away so it's not confusing. As the comment says, if we want to add more custom code then we can add a feature to have the menu change selection as we scroll, but I'm not sure I'd be in favor of that as it gets complicated when there are multiple items on the screen, or the scroll never gets that far.

For now, the selection on clicking an item is a nice feedback to the menu interaction.

> ```
> // TODO(petemill): If this wasn't a chromium module, we'd simply add a handler
> // for scrolling away, or have the menu change selection as we scroll.
> ```